packages/user-access-policy/src/rule/storage/benchmark/storageFactory/mongooseRulesStorageFactory.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright 2021-2024 Prosopo (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import mongoose, { type Model } from "mongoose";
+import type Rule from "../../../rule.js";
+import MongooseRulesStorage from "../../mongooseRulesStorage.js";
+import getMongooseRuleRecordSchema from "../../record/getMongooseRuleRecordSchema.js";
+import type RulesStorage from "../../rulesStorage.js";
+import type RulesStorageFactory from "./rulesStorageFactory.js";
+
+class MongooseRulesStorageFactory implements RulesStorageFactory {
+	async createRulesStorage(dbUrl: string): Promise<RulesStorage> {
+		const model = await this.createModel(dbUrl);
+
+		return new MongooseRulesStorage(model);
+	}
+
+	protected async createModel(db: string): Promise<Model<Rule>> {
+		const mongoConnection = await mongoose.connect(`mongodb://${db}`);
```

Comment:
This protocol is not always going to be correct. For replica sets the protocol is `mongodb+srv://`. There is code in our main codebase that handles this.